### PR TITLE
Fix: Docker - Git Not Found Error 

### PR DESCRIPTION
**Pull request by Lao-Liu233**
- Fix the problem of Docker compose: "ERROR: Cannot find command 'git'"
- Using Builder and Runtime to reduce the size of image

### CODE REVIEW

**Review by y-young — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+
+# Builder
+FROM python:${PYTHON_VERSION}-slim as builder
+
+WORKDIR /root
+COPY requirements.txt /root
+
+# Install requirements
+RUN apt update && apt install -y python3-pip git
+RUN pip install --prefix="/install" --no-warn-script-location -r requirements.txt
```
Comment (fzyzcjy) [inline Dockerfile:11]: uber-nit: Shall we rephrase it as

```
RUN apt update && apt install -y python3-pip git
COPY requirements.txt /root
RUN pip install --prefix="/install" --no-warn-script-location -r requirements.txt
```

Then, when requirements.txt changes, there is no need to apt update / apt install and we can reuse that cache

Comment (Misaka13514) [reply Dockerfile:11]: I agree with you.

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+ARG PYTHON_VERSION=3.8
+
+# Builder
+FROM python:${PYTHON_VERSION}-slim as builder
+
+WORKDIR /root
+COPY requirements.txt /root
+
+# Install requirements
+RUN apt update && apt install -y python3-pip git
```
Comment (Misaka13514) [inline Dockerfile:10]:
```suggestion
# Install requirements
RUN apt-get update && \
    apt-get install -y git && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
WORKDIR /root
COPY requirements.txt /root
```

File: Dockerfile (hunk 1 above)
Comment (y-young) [reply Dockerfile:11]: Fixed, thanks.

File: Dockerfile (hunk 2 above)
Comment (y-young) [reply Dockerfile:10]: Removed python3-pip.

This is a builder image, I guess doing clean or not will not affect the final image size, so I'll just leave it as it is to save build time.

Comment (Misaka13514) [reply Dockerfile:10]: It's better to use `apt-get` than `apt` in Dockerfile.

Comment (y-young) [reply Dockerfile:10]: Fixed.

**Pull request merged**